methods.py

```diff
@@ -83,6 +82,29 @@ def add_source_files(self, sources, files, allow_gen=False):
     return True
 
 
+def redirect_emitter(target, source, env):
+    """
+    Emitter to automatically redirect object/library build files to the `bin/obj` directory,
+    retaining subfolder structure. External build files will attempt to retain subfolder
+    structure relative to their environment's parent directory, sorted under `bin/obj/external`.
+    If `redirect_build_objects` is `False`, or an external build file isn't relative to the
+    passed environment, this emitter does nothing.
+    """
+    if not env["redirect_build_objects"]:
```

Comment:
Personally, I'd prefer if the `redirect_emitter` didn't query `env` for some string, but instead got an explicit argument passed to it that specified the target directory (e.g. `target_directory`, `None` for 'no redirect'). It would be agnostic to how the cli is set up, and just focus on the task of emitting obj files somewhere.

Reply:
That's a bridge to cross if/when the path is no longer hardcoded, which is currently outside the scope of this PR.